src/app/shared/related-items/related-items.component.ts

```diff
@@ -0,0 +1,239 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {ApiFileService} from '../../services/api-file.service';
+import * as FileSaver from 'file-saver';
+import setting from './related-items.component.setting.json';
+
+@Component({
+  selector: 'app-related-items',
+  templateUrl: './related-items.component.html',
+  styleUrls: ['./related-items.component.css']
+})
+export class RelatedItemsComponent implements OnInit {
+  @Input() record_id: string; // the record id used to retrieve particular record
+  @Input() source_type: string; // equal to the type of detail page, e.g. to list files in the dataset detail page, set to be dataset
+  @Input() target_type: string; // the related entities, e.g. to list files in the dataset detail page, set to be file
+  @Input() download_key: string; // if download not needed (nomrally not file), set to empty string, otherwise to the link attribute
+
+  records: any;
+  urls: string[] = [];
+  checked = false;
+  selected: Map<string, boolean> = new Map();
+
+  p = 1; // page number for html template
+  // to use this component, 4 steps:
+  // Step 1: add corresponding setting in the setting json file
+  // Step 2: in ngOnInit, add else if to retrieve the data
+  // Step 3: add this component into the detail page
+  // Step 4: make necessary adjustment (i.e. debugging) to the setting
+
+  constructor(private apiFileService: ApiFileService) {
+  }
+
+  ngOnInit() {
+    // Read in the initial column display settings
+    for (let i = 0; i < setting[this.source_type][this.target_type]['display'].length; i++) {
+      this.selected.set(setting[this.source_type][this.target_type]['display'][i], true);
+    }
+    for (let i = 0; i < setting[this.source_type][this.target_type]['all'].length; i++) {
```

Comment:
This should be simplified to
for (const column of setting[this.source_type][this.target_type]['all']) {
if (!this.selected.has(column))
this.selected.set(column, false)
}

Reply:
done in 083df0d